File: globaldefensesurveillancetracking_ui/src/services/NationService.js
import http from "../http-common";

class NationService {
  getAllNations(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/nation/nations`, searchDTO);
  }

  get(nationId) {
    return this.getRequest(`/nation/${nationId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/nation?field=${matchData}`, null);
  }

  addNation(data) {
    return http.post("/nation/addNation", data);
  }

  update(data) {
  	return http.post("/nation/updateNation", data);
  }
  
  uploadImage(data,nationId) {
  	return http.postForm("/nation/uploadImage/"+nationId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new NationService();

File: src/main/java/com/globaldefensesurveillancetracking/service/NationService.java
package com.globaldefensesurveillancetracking.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.globaldefensesurveillancetracking.domain.Nation;
import com.globaldefensesurveillancetracking.dto.NationDTO;
import com.globaldefensesurveillancetracking.dto.NationSearchDTO;
import com.globaldefensesurveillancetracking.dto.NationPageDTO;
import com.globaldefensesurveillancetracking.dto.NationConvertCriteriaDTO;
import com.globaldefensesurveillancetracking.service.GenericService;
import com.globaldefensesurveillancetracking.dto.common.RequestDTO;
import com.globaldefensesurveillancetracking.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface NationService extends GenericService<Nation, Integer> {

	List<Nation> findAll();

	ResultDTO addNation(NationDTO nationDTO, RequestDTO requestDTO);

	ResultDTO updateNation(NationDTO nationDTO, RequestDTO requestDTO);

    Page<Nation> getAllNations(Pageable pageable);

    Page<Nation> getAllNations(Specification<Nation> spec, Pageable pageable);

	ResponseEntity<NationPageDTO> getNations(NationSearchDTO nationSearchDTO);
	
	List<NationDTO> convertNationsToNationDTOs(List<Nation> nations, NationConvertCriteriaDTO convertCriteria);

	NationDTO getNationDTOById(Integer nationId);







}





